packages/shell/esm-app-shell/src/ui/breadcrumbs.tsx

```diff
@@ -26,6 +30,40 @@ function getParams(path: string, matcher: RegExp) {
   return [];
 }
 
+interface CustomBreadcrumbItemProps {
+  bc: BreadcrumbRegistration;
+  params: any;
+}
+
+export const CustomBreadcrumbItem: React.FC<CustomBreadcrumbItemProps> = ({
+  bc,
+  params,
+}) => {
+  const [title, setTitle] = useState("");
+
+  useEffect(() => {
+    if (typeof bc.settings.title === "function") {
+      if (bc.settings.title?.constructor?.name === "AsyncFunction") {
+        (bc.settings.title(params) as Promise<string>).then((res) =>
+          setTitle(res)
+        );
+      } else {
+        setTitle(bc.settings.title(params) as string);
+      }
```

Comment:
I didn't know about the `AsyncFunction` constructor, that's cool! However I think a more robust way of dealing with this will be

```suggestion
      Promise.resolve(bc.settings.title(params)).then((res) =>
        setTitle(res)
      );
```